### PR TITLE
Don't force the user to install a puppetmaster

### PR DESCRIPTION
**Pull request by GregSutcliffe**
There's probably more worth fixing in this story, but at least with this change I can create hosts with just a TFTP proxy...

### CODE REVIEW

File: app/helpers/hosts_and_hostgroups_helper.rb [hunk 1]
```diff
@@ -53,27 +53,27 @@ def os_ptable
...
-    # if we are not using provisioning, not much point in presenting the CA option (assuming your CA is already set otherwise)
-    return unless SETTINGS[:unattended]
-    proxies = SmartProxy.joins(:features).where(:features => { :name => "Puppet CA" })
+    # Don't show this if we have no CA proxies, otherwise always include blank
+    # so the user can choose not to sign the puppet cert on this host
+    proxies = SmartProxy.puppetca_proxies
+    return if proxies.count == 0
     select_f f, :puppet_ca_proxy_id, proxies, :id, :name,
-             { :include_blank => proxies.to_a.count > 1 },
+             { :include_blank => true },
```
Comment (ohadlevy) [inline app/helpers/hosts_and_hostgroups_helper.rb:65]: why?

**Comment by ohadlevy** [comment]
While on a simliar topic, does it make sense (now that we detect the puppet proxy that a report/facts comes from) to assign it to the host based on request? /cc @domcleal

**Comment by ohadlevy** [comment]
@abenari please review, @GregSutcliffe I assume we need the same for hostgroup too?

File: app/helpers/hosts_and_hostgroups_helper.rb (hunk 1 above)
Comment (domcleal) [inline app/helpers/hosts_and_hostgroups_helper.rb:65]: so Puppet truly is optional on any host, even if you have a puppet proxy.  This way you could use Puppet for various hosts, but add some non-Puppet device as a host.

**Comment by domcleal** [comment]
@ohadlevy that'd make some sense, but we can only really do the puppetmaster, not the CA.  A followup feature I think.

**Comment by domcleal** [comment]
Actually, what you're suggesting there @ohadlevy is a fix for [#1830](http://projects.theforeman.org/issues/1830).

File: app/helpers/hosts_and_hostgroups_helper.rb (hunk 1 above)
Comment (ohadlevy) [inline app/helpers/hosts_and_hostgroups_helper.rb:65]: i understand, but if you have a puppetmaster, i would suggest we still alow it as a default, and let the user select not using it.

also... if we unselect it, we need some JS to hide classes / parateres

**Comment by ohadlevy** [comment]
@domcleal i know ;)

File: app/helpers/hosts_and_hostgroups_helper.rb (hunk 1 above)
Comment (GregSutcliffe) [inline app/helpers/hosts_and_hostgroups_helper.rb:65]: Include_blank is just so you can unset puppet on a host, I have no problem with setting a puppetmaster as a default. You are correct in saying there is some JS issues - I've already hidden the Classes tab, but if you select a hostgroup, classes from the group are still applied. It's harmless though, as classes will never be run on a system with no puppet master ;)

**Comment by GregSutcliffe** [comment]
@ohadlevy there's certainly more fixes around hostgroups etc that would clean up cosmetic issues with assigning classes/class params to hosts/hostgroups with no puppetmaster, but as it's all cosmetic, I think we should catch the obvious ones and then get it merged before code freeze on Thurs. We can bug fix anything we missed as we go forward into RC-land

**Comment by ohadlevy** [comment]
merged as a92cdd4 and f764065 (with minor changes)

**Pull request closed without merging**